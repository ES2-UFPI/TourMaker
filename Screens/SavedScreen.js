import React, { Component } from 'react';
import { View, Text, Modal } from 'react-native';
import CustomTable from './ScreensModules/CustomTable';
import TableButton from './ScreensModules/TableButton';
import Styles from './Styles';
import CustomMapView from './ScreensModules/CustomMapView';


import AsyncStorageFunctions from '../APIs/AsyncStorage';
class SavedScreen extends Component {
    constructor(props) {
        super(props)
        this.state = {
            listaRoteiros: this.recoverFav(),
            tableHead: ["Titulo","Opções"],
            userLocation: null
        }
    }

    recoverFav(){
        AsyncStorageFunctions.retrieveData("favoritos",(result)=>{
            this.setState({
                listaRoteiros: result
            })
            })
    }

    deleteFav(index1){
        var newListaRoteiros= []
        this.state.listaRoteiros.forEach((element,index) => {
            if (index1 != index){
                newListaRoteiros.push(element)
            }   
        });
        this.setState({
            listaRoteiros: newListaRoteiros
        })
        AsyncStorageFunctions.storeData(newListaRoteiros, "favoritos")
    }
    
    useFav(index){
        const paramsRota = {
            listaPDI: this.state.listaRoteiros[index].rota,
            userLocation: this.state.userLocation
        }
        var routeLength = this.props.navigation.dangerouslyGetState().routes.length
        var previousScreen = this.props.navigation.dangerouslyGetState().routes[routeLength - 2].name
        if (previousScreen == 'Home') {
            this.props.navigation.push('GerenciamentoRoteiro', paramsRota)
        } else {
            this.props.navigation.pop()
        }
    }


    render() {  
        var tableData = []
        if (this.state.listaRoteiros!= null){
            this.state.listaRoteiros.forEach((element, index) => {
                
                tableData.push([
                    element.titulo,
                    TableButton(index, this.deleteFav.bind(this), 'Excluir'),
                    TableButton(index, this.useFav.bind(this), 'Gerenciar')
                ])
            });
        } 
        return(
            <View style={Styles.Screen}>
            <Text>Gereciamento de Roteiro</Text>
                <View style={Styles.Container}>
                    <CustomTable
                        tableHead={this.state.tableHead}
                        tableData={tableData}
                    />
                <CustomMapView
                style={{height: 0}}
                markers={this.state.listaPDI}
                UserLocation={(location) => {
                    this.setState({
                        userLocation: location
                    })
                }}
            />
                </View>
                
            </View>
        )
    }
}

export default SavedScreen;