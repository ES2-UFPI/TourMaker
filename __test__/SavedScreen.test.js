import React from 'react';
import renderer from 'react-test-renderer';

import SavedScreen from '../Screens/SavedScreen';


it('renders correctly', () => {
    const tree = renderer.create(<SavedScreen route={{ params: {} }} />).toJSON();
    expect(tree).toMatchSnapshot();
});