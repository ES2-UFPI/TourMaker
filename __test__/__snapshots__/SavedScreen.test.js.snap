// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text>
    Gereciamento de Roteiro
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "padding": 16,
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "borderBottomWidth": 0,
            "borderColor": "transparent",
            "borderLeftWidth": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            0,
            Object {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#1abc9d",
              "width": 400,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Titulo
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "transparent",
                "borderRightWidth": 0,
                "borderTopWidth": 0,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Opções
          </Text>
        </View>
      </View>
      <RCTScrollView>
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "yellow",
          "height": 0,
          "justifyContent": "flex-start",
          "width": "100%",
        }
      }
    >
      <AIRMap
        followsUserLocation={false}
        handlePanDrag={false}
        initialRegion={null}
        onChange={[Function]}
        onLayout={[Function]}
        onMapReady={[Function]}
        onMarkerPress={[Function]}
        onRegionChangeComplete={[Function]}
        onUserLocationChange={[Function]}
        region={null}
        showsUserLocation={false}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</View>
`;
